js/blockchain.js

```diff
@@ -27,205 +28,500 @@ if (typeof XPCNativeWrapper === 'function') {
+                    cursor = null;
+                }
+                return jsonData;
+            });
+        }
+        const _getUserData = (jsonData) => {
+            if (jsonData.hasOwnProperty('ids')) {
+                // split array of ids into 100 user chunks
+                let i = 0;
+                let chunks = [];
+                while (i * 100 < jsonData.ids.length) {
+                    let slice = jsonData.ids.slice(i * 100, i * 100 + 100);
+                    chunks.push(slice.join(","));
+                    i++;
+                }
+                chunks = chunks.map((element) => {
+                    let url = 'https://api.twitter.com/1.1/users/lookup.json?include_entities=true'
+                    return this._makeRequest({
+                        url: url,
+                        headers: {
+                            'content-type': 'application/x-www-form-urlencoded'
+                        },
+                        body: `user_id=${element}`,
+                        method: 'POST'
+                    })
+                })
+                return Promise.all(chunks).then((values) => {
+                    values = values.map((users) => {
+                        return JSON.parse(users);
+                    })
+                    return {
+                        users: values.flat(1)
+                    };
+                })
+            } else {
+                throw new Error('no ids');
+            }
+        }
+        const _processData = (jsonData) => {
+            let scratch_usersFound = 0;
+            let scratch_usersSkipped = 0;
+            let scratch_usersAlreadyBlocked = 0;
+            if (jsonData) {
+                var users = jsonData.users
+                    .filter((element) => {
+                        if (element.following || element.screen_name in protectedUsers) {
+                            scratch_usersSkipped++;
+                            return false;
+                        }
+                        return true;
+                    })
+                    .filter((element) => {
+                        if (element.blocking) {
+                            scratch_usersFound++;
+                            scratch_usersAlreadyBlocked++;
+                            return false;
+                        }
+                        return true;
+                    })
+                    .filter((element) => {
+                        return element.screen_name != null;
+                    })
+                    .map((element) => {
+                        scratch_usersFound++;
+                        return {
+                            username: element.screen_name,
+                            id: element.id_str
+                        };
+                    });
+                usersFound += scratch_usersFound;
+                usersSkipped += scratch_usersSkipped;
+                usersAlreadyBlocked += scratch_usersAlreadyBlocked;
```

Comment:
maybe forgot to update `totalCount` here?

Reply:
good point... i've changed the way the code works so that totalCount is not set until after the blocker is complete. this simplified a lot of logic for when the blocker is supposed to run.